src/screens/AddSiteScreen.js

```diff
@@ -125,24 +142,45 @@ class AddSiteScreen extends Component {
         this.props.logUserIntoFacebook({navigate});
     };
 
+    renderWaitingOnConnectionIfNecessary() {
+        const {pendingUploadSites} = this.props;
+        const {infoTextStyle, errorTextStyle, waitingSignalRowStyle, leftMargin} = styles;
+
+        const thereAreSitesToUpload = this.thereAreSitesToUpload({pendingUploadSites});
+        const connectionIsStrongEnough = this.connectionIsStrongEnough(this.props);
+
+        if (thereAreSitesToUpload && !connectionIsStrongEnough) {
+            return (
+                <View style={waitingSignalRowStyle}>
+                    <ActivityIndicator size='large'/>
+                    <Text style={[infoTextStyle, errorTextStyle, leftMargin]}>{pending_upload_sites_waiting}</Text>
+                </View>
+            );
+        } else {
+            return null;
+        }
+    }
+
     renderPendingUploadSites() {
         const {pendingUploadSites} = this.props;
-        const {headerTitleStyle} = styles;
+        const {headerTitleStyle, infoTextStyle} = styles;
 
         if (pendingUploadSites.length > 0) {
             const {navigation: {navigate}} = this.props;
             const {listContainerStyle} = styles;
 
             return (
-                <View style={listContainerStyle}>
+                <Card style={listContainerStyle}>
                     <Text style={headerTitleStyle}>{pending_upload_sites_header}</Text>
+                    <Text style={infoTextStyle}>{pending_upload_sites_description}</Text>
+                    {this.renderWaitingOnConnectionIfNecessary()}
```

Comment:
Changed this component to a Card, since there can be more than one Pending list on the screen at a time, added descriptive text of what it is, and a conditional for the helper that gives info on poor connection.